### ISSUE TITLE
When installing the cookbook I get ERROR: ArgumentError: "\x80\x00\x00\x00d\x9E\xD3\xD9" is not an octal string

### ISSUE THREAD

**Issue by kmf** [sous-chefs/line#92]
Hi there
I'm getting a weird error when I try and install a cookbook
Using chef-dk, version 2.5.3-1

![2018-05-18-002](https://user-images.githubusercontent.com/78700/40221013-70a4e644-5a7b-11e8-87d1-f59e6c32703b.png)

**Comment by damacus** [comment]
This sounds like a ChefDK problem. Can you try a `berks install line` instead? 

For reference you can 
- create a wrapper cookbook
- depend on line in that cookbook
- run kitchen test 

And it'll get all the dependencies you want without manually installing them using `knife`.

**Comment by kmf** [comment]
@damacus 
```
Resolving cookbook dependencies...
Fetching 'blah' from source at .
Fetching cookbook index from https://supermarket.chef.io...
Using iptables (4.3.4)
Installing line (1.1.1)
Using blah (0.1.0) from source at .
Using openssh (2.6.3)
Using sudo (5.4.0)
Using users (5.3.1)
/opt/chefdk/embedded/lib/ruby/site_ruby/2.4.0/rubygems/package/tar_header.rb:128:in `strict_oct': "\x80\x00\x00\x00d\x9E\xD3\xD9" is not an octal string (ArgumentError)
	from /opt/chefdk/embedded/lib/ruby/site_ruby/2.4.0/rubygems/package/tar_header.rb:108:in `from'
	from /opt/chefdk/embedded/lib/ruby/site_ruby/2.4.0/rubygems/package/tar_reader.rb:59:in `each'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/mixlib-archive-0.4.1/lib/mixlib/archive/tar.rb:27:in `block in extract'
	from /opt/chefdk/embedded/lib/ruby/site_ruby/2.4.0/rubygems/package/tar_reader.rb:29:in `new'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/mixlib-archive-0.4.1/lib/mixlib/archive/tar.rb:118:in `reader'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/mixlib-archive-0.4.1/lib/mixlib/archive/tar.rb:25:in `extract'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/mixlib-archive-0.4.1/lib/mixlib/archive.rb:33:in `extract'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/community_rest.rb:16:in `unpack'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/community_rest.rb:102:in `download'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/downloader.rb:70:in `try_download'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/downloader.rb:40:in `block in download'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/downloader.rb:39:in `each'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/downloader.rb:39:in `download'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/installer.rb:108:in `install'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/berkshelf-6.3.1/lib/berkshelf/installer.rb:181:in `block (2 levels) in install_from_universe'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/safe_task_executor.rb:24:in `block in execute'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `block in synchronize'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/synchronization/mri_lockable_object.rb:38:in `synchronize'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/safe_task_executor.rb:19:in `execute'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/ivar.rb:170:in `safe_execute'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/future.rb:52:in `block in execute'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	from /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/concurrent-ruby-1.0.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'

```

**Comment by kmf** [comment]
Changed to chefdk 3.0.36-1
worked.

```
Installing line to /home/kmf/blah/code/chef/blah-chef-repo/cookbooks
Downloading line from Supermarket at version 1.1.1 to /home/kmf/blah/code/chef/blah-chef-repo/cookbooks/line.tar.gz
Cookbook saved: /home/kmf/blah/code/chef/blah-chef-repo/cookbooks/line.tar.gz
Removing pre-existing version.
Uncompressing line version 1.1.1.
Removing downloaded tarball
1 files updated, committing changes
Creating tag cookbook-site-imported-line-1.1.1
Already up-to-date.
Cookbook line version 1.1.1 successfully installed
```

**Comment by damacus** [comment]
Can you try the latest/current ChefDK please?

IIRC, this is exactly what the newly package libarchive will fix.

https://downloads.chef.io/chefdk/current

**Comment by damacus** [comment]
haha, thanks. Have a good day!

**Comment by kmf** [comment]
@damacus :+1:

**Comment by eyespies** [comment]
I ran into this error with this cookbook when running Berkshelf in my CI/CD process using bundler and a Rakefile. It worked fine when I installed ChefDK and use the CLI `berks` command, but always failed when I ran `bundle exec rake lint spec`. 

I finally tracked down that I had to `apt-get install libarchive13` and include in my *Gemfile* `gem ffi-libarchive`, otherwise the mixlib-archive Gem falls back to using Tar module instead of libarchive and Tar on Linux doesn't work well with archives built on a Mac without special options on the Mac side.
